app/controllers/posts_controller.rb

```diff
@@ -19,9 +19,9 @@ def create
     @post = current_user.post.build(permit_post)
     if @post.save
       flash[:success] = 'The Post you created was a Success!'
-      redirect_to posts_path(@post)
+      redirect_to posts_path
     else
-      flash[:error] = @post.errors.full_messages
+      flash.now[:error] = 'You have to write something to  publish'
```

Comment:
- [ ] You still have the `redirect_to` in in case the post is not saved. `redirect_to` will start a new request losing all data that the user has entered before, that's why we need `render` here. Please take a look at [this](http://tutorials.jumpstartlab.com/topics/controllers/render_and_redirect.html)